compiler/expressions.go

```diff
@@ -1130,12 +1136,17 @@ func (c *funcContext) translateImplicitConversion(expr ast.Expr, desiredType typ
 			// wrap JS object into js.Object struct when converting to interface
 			return c.formatExpr("new $jsObjectPtr(%e)", expr)
 		}
+
+		switch exprType.Underlying().(type) {
```

Comment:
Can this maybe be solved instead by making `translateImplicitConversionWithCloning` look at `c.p.TypeOf(expr)` instead of `desiredType`? This would also catch the case of "array/struct to interface".